command/agent/http.go

```diff
@@ -193,6 +194,58 @@ func NewHTTPServers(agent *Agent, config *Config) ([]*HTTPServer, error) {
 		srvs = append(srvs, srv)
 	}
 
+	// Initialize the API socket if configured
+	startAPISocket := func() {
```

Comment:
I would prefer a top level func that takes a config and returns an error just because this func is already so huge and only readable because it's the same code over and over slight slight variations. nbd though

Reply:
I'll drop a TODO here so that I can come back to is but it doesn't stall the world.